### PR TITLE
[#40][Backend] Write test for google scraper service

### PR DESCRIPTION
**Pull request by sanG-github**
## What happened

We don't have any tests for this service so we need to write tests to ensure the service always runs as expected.
Add gem `VCR` to record a HTTP request

## Insight

As the issue mentioned about, the tests will cover 2 main things:
- Validate parameters when passing to the service: result_id. In the worth case, it's will log and the developer can aware.
- Store the results and then trigger a new worker to push real-time data to the user's view.

## Proof Of Work

image

### CODE REVIEW

File: spec/services/crawl_google_data_service_spec.rb [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+        keyword = 'nimble'
+        result = create :result, keyword: keyword, status: :fetching
+        subject = described_class.new(result_id: result.id)
+        worker_jobs = PushResultWorker.jobs
+
+        VCR.use_cassette("nimble_result") do
+          subject.call
+
+          expect(worker_jobs.size).to eq(1)
+          expect(worker_jobs.first['args'][0]).to eq(result.user.id)        end
```
Comment (tyrro) [inline spec/services/crawl_google_data_service_spec.rb:64]:
```suggestion
          expect(worker_jobs.first['args'][0]).to eq(result.user.id)
        end
```

File: spec/services/crawl_google_data_service_spec.rb [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+      end
+    end
+
+    context 'when fetch result successfully' do
+      it 'updates the result record' do
+        keyword = 'nimble'
+        result = create :result, keyword: keyword, status: :fetching
+        subject = described_class.new(result_id: result.id)
+
+        VCR.use_cassette("nimble_result") do
```
Comment (tyrro) [inline spec/services/crawl_google_data_service_spec.rb:32]: not sure, maybe `search_results/nimble.yml` might be a good idea? A dedicated folder for storing `search_results`?

**Review by tyrro — APPROVED** [review]

File: spec/services/crawl_google_data_service_spec.rb (hunk 2 above)
Comment (sanG-github) [reply spec/services/crawl_google_data_service_spec.rb]: Thanks for nice suggestion. I added it in b2d4d44 ✅

**Review by rosle — APPROVED** [review]

File: spec/services/crawl_google_data_service_spec.rb [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+          subject.call
+
+          expect(worker_jobs.size).to eq(1)
+          expect(worker_jobs.first['args'][0]).to eq(result.user.id)
+        end
+      end
+    end
+
+    context 'when fetch result failed' do
+      it 'rollbacks from start' do
```
[the remark this answers is not in the document]
Comment (sanG-github) [reply spec/services/crawl_google_data_service_spec.rb]: Changed it in aaf00e9 ✅

**Pull request merged**